pkg/backends/datadog/datadog.go

```diff
@@ -38,7 +38,7 @@ const (
 	// defaultMetricsPerBatch is the default number of metrics to send in a single batch.
 	defaultMetricsPerBatch = 1000
 	// maxResponseSize is the maximum response size we are willing to read.
-	maxResponseSize     = 10 * 1024
+	maxResponseSize     = 1024
```

Comment:
I reduced this (and in NR) because the response is generally minimal, and we log it.  We shouldn't be logging 10k of data, especially since failures are likely to cluster, so we could end up with hundreds of 10k logs.  There's an argument to be made for "read up to 10k, log up to 1k", just to ensure we drain the connection if possible, without reading to infinity, but we can add that if we need it.